hack/rp-dev/rp_funcs.sh

```diff
@@ -0,0 +1,197 @@
+  echo -e "Success step 5 ✅ - Hive has been installed\n"
+}
+
+mirror_images() {
+  export DST_ACR_NAME=${AZURE_PREFIX}aro
+  export SRC_AUTH_QUAY=$(echo $USER_PULL_SECRET | jq -r '.auths."quay.io".auth')
+  export SRC_AUTH_REDHAT=$(echo $USER_PULL_SECRET | jq -r '.auths."registry.redhat.io".auth')
+  export DST_AUTH=$(echo -n '00000000-0000-0000-0000-000000000000:'$(az acr login -n $DST_ACR_NAME --expose-token | jq -r .accessToken) | base64 -w0)
+  docker login -u 00000000-0000-0000-0000-000000000000 -p "$(echo $DST_AUTH | base64 -d | cut -d':' -f2)" "$DST_ACR_NAME.azurecr.io"
+  echo "Success step 6a ✈️ 🏷️ - Login to ACR"
+
+  if [[  "${SKIP_DEPLOYMENTS}" == "false" ]] || ! check_acr_repos ${AZURE_PREFIX}-global ; then
+    make go-verify
+    echo "Success step 6b - Add Vendor directory"
+
+    go run -tags containers_image_openpgp,exclude_graphdriver_btrfs ./cmd/aro mirror latest
+    echo "Success step 6c ✈️ 📦 - Mirror OCP images"
+
+    mdm_image_tag=$(get_digest_tag "MdmImage")
+    mdm_repo=$(echo $mdm_image_tag | sed -n 's|.*/\([^/]*\/[^:]*\):.*|\1|p')
+    mdsd_image_tag=$(get_digest_tag "MdsdImage")
+    mdsd_repo=$(echo $mdsd_image_tag | sed -n 's|.*/\([^/]*\/[^:]*\):.*|\1|p')
+    import_geneva_image $mdm_repo $mdm_image_tag
+    import_geneva_image $mdm_repo $mdm_image_tag
+    echo "Success step 6d ✈️ 📦 - Import MDM and MDSD to ACR"
+
+    make publish-image-aro-multistage
+    echo "Success step 6e ✈️ 📦 - Build, push and import to ARO image to ACR"
+
+    fluentbit_image_tag=$(get_digest_tag "FluentbitImage")
+    copy_digest_tag $PULL_SECRET "arointsvc" $DST_ACR_NAME $fluentbit_image_tag
+    echo "Success step 6f ✈️ 📦 - Copy Fluenbit image to ACR"
+  else
+    echo "⏩📋 Skip mirroring repos to ACR $DST_ACR_NAME" 
+  fi
+  echo -e "Success step 6 ✅ - Mirror repos to ACR $DST_ACR_NAME\n"
+}
+
+prepare_RP_deployment() {
+  export PARENT_DOMAIN_NAME=osadev.cloud
+  export GLOBAL_RESOURCEGROUP=${AZURE_PREFIX}-global
+
+  for DOMAIN_NAME in ${AZURE_PREFIX}-clusters.$PARENT_DOMAIN_NAME ${AZURE_PREFIX}-rp.$PARENT_DOMAIN_NAME; do
+    CHILD_DOMAIN_PREFIX="$(cut -d. -f1 <<<$DOMAIN_NAME)"
+    echo "########## Creating NS record to DNS Zone $CHILD_DOMAIN_PREFIX ##########"
+    az network dns record-set ns create \
+      --resource-group "$PARENT_DOMAIN_RESOURCEGROUP" \
+      --zone "$PARENT_DOMAIN_NAME" \
+      --name "$CHILD_DOMAIN_PREFIX" >/dev/null
+    for ns in $(az network dns zone show \
+      --resource-group "$GLOBAL_RESOURCEGROUP" \
+      --name "$DOMAIN_NAME" \
+      --query nameServers -o tsv); do
+      az network dns record-set ns add-record \
+      --resource-group "$PARENT_DOMAIN_RESOURCEGROUP" \
+      --zone "$PARENT_DOMAIN_NAME" \
+      --record-set-name "$CHILD_DOMAIN_PREFIX" \
+      --nsdname "$ns" >/dev/null
+    done
+  done
+  echo "Success step 7a ✅ - Update DNS Child Domains"
+
+  check_and_import_certificates
+  echo "Success step 7b ✅ - Update certificates in keyvault"
```

Comment:
1. You can test the return code of `check_and_import_certificates` to see if it was successful, before logging that it was.
2. Use `log`